### PR TITLE
Code Quality Fix - Constructors call non overridable method

### PR DESCRIPTION
**Pull request by civanyp**
DevFactory is committed to improving code quality and advancing open source. Our team contributes their time to researching and identifying areas for code quality improvement in open source projects. We identified a fix for NanoHTTPD.
This pull request is focused on resolving occurrences of Sonar rule squid:S1699 - “Constructors should only call non-overridable methods”. You can find more information about the issue here: https://dev.eclipse.org/sonar/rules/show/squid:S1699
Please let me know if you have any questions. There is more to come.

Christian Ivan
DevFactory - Code Cleanup Team

### CODE REVIEW

**Comment by LordFokas** [comment]
Forgive me for my harshness but you're not fixing anything. NanoHTTPD (and NanoWSD) are meant to be integrated into other software, which means everything being overrideable is a feature. If the person using it has a reason to override methods, even those called in constructors, he can, and if not the default works pretty well. In making methods final you are instead breaking that feature which we have been discussing over time how to keep and maintain properly.

In fact, NanoWSD is only possible because it can easily override the way NanoHTTPD interprets the HTTP protocol by default.

This goes out to both you and @javedanski: I'm just another contributor in this project, which I have been using for about 2 years now, and I don't mean to discourage you or anything, but I honestly think you have no idea what you're doing. I mean, I've checked your github accounts (which are fairly new) and all I see you doing is hopping from repo to repo, running their code against some automated tool that compares code against some other team's standards -- standards that most projects **don't** need -- and then shoot in every direction: whatever the tool tells you it's wrong you try to fix it and rush a PR titled as a "code quality fix", when you don't even know what is happening or even worse **WHY** it is an issue. That seriously troubles me: how can you fix something if you don't know why it's broken?? In all your _shoot-in-every-direction-in-hopes-to-kill-something_ efforts you seem to be harming the project more than you're helping (and that's why you should focus on the project and not on whatever tools you use).

For example:
- This (#258) breaks more than it fixes.
- #259 adds a nullcheck, but knowing the project if in any situation there's a null coming in in that location then you have bigger problems at the data source. If the data source is fine you'll never get a null there because more data is expected after that string.
- #260 as I mentioned in the issue itself, is a "fix" that uses CPU power and memory to fix a non-issue.
- #256 & #257 were already rejected (I haven't read, but can only assume for good reasons given what I've seen so far)
- #255 was accepted, but it **really** shouldn't. Honestly, just like #259, if you have nulls there then you have bigger problems somewhere else. Just another case of you treating **and hiding** the symptoms and not caring for the true cause.

**TL;DR:** If you're going to contribute to a project, take the time to really understand how it works internally, how it all flows. Otherwise you're just wasting your time and the team's time, and NanoHTTPD's team is already spread too thin and with little time to waste on such things.

@ritchieGitHub, @elonen: for the reasons stated above, I would like to see #255 reverted. "Yoda conditions" are _fugly_ and in this case they're hiding nulls coming from elsewhere that should never be null, therefore treating the symptoms and not the cause, and making issues that may arise from it much harder to track down. It's up to you guys, but I've presented my reasoning.

**Comment by ritchieGitHub** [comment]
@LordFokas I concur for 90% with your statement! These code analysis are good tools but have to be tuned to the project, and fixing a rule is in most cases not just adding a is-null somewhere, but the why question must be solved first and that can only be answered after a more thorough code review.  As for #255, I think that's one of the "correct" rules, and would keep the PR. @elonen ?

**Comment by ritchieGitHub** [comment]
@LordFokas @elonen lets discuss this further in this thread even when I close this issue because in our case it is completely wrong.
